### ISSUE TITLE
UI not updating as expected in functional component

### ISSUE THREAD

**Issue by injas427** [facebook/react-native#25968]
React Native version:

```
    OS: macOS 10.14.5
    CPU: (4) x64 Intel(R) Core(TM) i5-3210M CPU @ 2.50GHz
    Memory: 632.37 MB / 10.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 10.15.3 - /usr/local/bin/node
    Yarn: 1.16.0 - /usr/local/bin/yarn
    npm: 6.4.1 - /usr/local/bin/npm
  SDKs:
    iOS SDK:
      Platforms: iOS 12.2, macOS 10.14, tvOS 12.2, watchOS 5.2
    Android SDK:
      API Levels: 23, 27, 28, 29
      Build Tools: 27.0.3, 28.0.3, 29.0.1
  IDEs:
    Android Studio: 3.4 AI-183.6156.11.34.5692245
    Xcode: 10.2.1/10E1001 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.8.6 => 16.8.6 
    react-native: ^0.60.4 => 0.60.4 
  npmGlobalPackages:
    react-native-cli: 2.0.1
    react-native-create-library: 3.1.2
    react-native-git-upgrade: 0.2.7
```

### Actual behavior:

My app supports multiple language (English and Arabic). I'm handling localization using `redux-i18n` library. Language switching is working fine and the language stored in app changes. But only in functional component the UI is not re-rendering when the language changed.

I'm flipping direction when the language changes. I use `flexDirection` for it. 

For example:

```
flexDirection: `row${lang=="en"?"":"-reverse"}`
```

But `flexDirection` not updating when there is change in language.  The strings are getting changed. 

I've bottom tabbar which is class based component and it works fine when language changes

![UI render](https://user-images.githubusercontent.com/14922455/62602741-0f5fd780-b912-11e9-9c4c-ba55aff48f6c.gif)

### Expected behavior:

When the language changes, the UI should accept the change and update the view accordingly

**Comment by injas427** [comment]
```
<FlatList
horizontal
showsHorizontalScrollIndicator={false}
data={shopDetailState.categories}
keyExtractor={(item, index) => index.toString()}
extraData={shopDetailState.categories}
renderItem={({ item, index }) => {

return <TouchableOpacity style={{ borderBottomColor: Colors.primaryColor, padding: Width * 0.02, marginHorizontal: Width * 0.004, marginBottom: Width * 0.05, borderBottomWidth: shopDetailState.activeCategory == item.id ? Width * 0.002 : 0 }} onPress={() => dispatch({ type: "SET_ACTIVE_CATEGORY", payload: item.id })} >

<Text style={[shopDetailState.activeCategory == item.id ? CommonStyles.fontBold : CommonStyles.fontRegular, CommonStyles.font045, { color: shopDetailState.activeCategory == item.id ? Colors.primaryColor : Colors.detailPageAbout }]} >{item.name}</Text>
</TouchableOpacity>
}}

/>
```

I've this component and when I click on the button the current values gets updated but the style for `<Text>` component isn't taking the new change and I'm unable to notify the user about the same. But the `borderBottomWidth` accepting new change

**Comment by victorlevasseur** [comment]
FlatList doesn't update its content at each render. It only updates if the list (itself) changes (not its content if it's still the same list object) or if *extraData* changes. I think you need to addthe `shopDetailState.activeCategory` to the `extraData` prop of `FlatList`.

**Comment by injas427** [comment]
@victorlevasseur I've even tried `textAlign` prop for simple text (outside `FlatList`) and it didn't work either

**Comment by injas427** [comment]
The issue resolved in `0.60.5`
